### DISCUSSION TITLE
Temporary containers interoperability?

### DISCUSSION THREAD

**Discussion by ghost** [tridactyl/tridactyl#3893] [Q&A]
I have temporary containers enabled, but I can't seem to find a convenient way to have tridactyl open a new tab in a temporary container. If I use `tabopen`, it will open the tab in the container the tab I had active. The workaround is:

- open a new tab (CTRL+T)
- SHIFT+T
- CTRL+,
- `:tabopen`
- close the empty new tab

This is pretty inconvenient. Is there a way to get what I want?

**Comment by bovine3dom** [comment]
Have you tried `:set tabopencontaineraware false`?

**Reply by ghost** [reply]
Thanks for the suggestion. Just tried it, didn't seen to make a difference.

**Reply by bovine3dom** [reply]
How about `:tabopen -c firefox-default`?

Edit: if that doesn't work, I think there's a Tridactyl bug here https://github.com/tridactyl/tridactyl/blob/da9bea7edf63dd3ec24bd44c120cb72d6530ee18/src/lib/autocontainers.ts#L260

**Reply by ghost** [reply]
Hmm that does work, but it does not work when combined with search.

**Reply by ghost** [reply]
Note: this seems to work somewhat if you set Global -> Navigation -> Target Domain to "Different from Tab Domain & Subdomains" in temporary containers. The only issue is now `:tabopen search` doesn't work, forcing me to use CTRL+K. As good as it's going to get I guess.

**Reply by bovine3dom** [reply]
> Hmm that does work, but it does not work when combined with search.

Did you follow the instructions that the error message gave?

```# controller in excmd: Error: Firefox search engines do not support containers or background tabs in FF >80. `:set searchengine google` or see issue https://github.com/tridactyl/tridactyl/issues/2695```

**Reply by ghost** [reply]
No, I ended up setting `tabopencontaineraware` to `false` and fiddling with temporary container's isolation settings. I don't think this is a tridactyl issue anymore, but I will draft up an answer about what I had to change to get this to work once I figure out this last issue.

**Comment by ghost** [comment]
OK, I think I ironed out the kinks. Assume you have temporary containers and multi account containers installed and some permanent containers setup.

For temporary containers, configure Isolation -> Global -> Navigation to "Different from Tab Domain & Subdomains". Also configure mouse clicks if you want. In Advanced -> Isolation select "Instead of creating a new tab replace the current tab in case of Isolation". For multi account containers, make sure that you do **not** have "limit to designated sites". 

My `tridactylrc` looks like this:

```
autocmd DocStart (https?://)www.reddit.com js tri.excmds.urlmodify("-t", "www.reddit.com", "teddit.net")                                               
autocmd DocStart (https?://)old.reddit.com js tri.excmds.urlmodify("-t", "old.reddit.com", "teddit.net")                                               
autocmd DocStart (https?://)twitter.com js tri.excmds.urlmodify("-t", "twitter.com", "nitter.42l.fr")                                                  
autocmd DocStart (https?://)mobile.twitter.com js tri.excmds.urlmodify("-t", "mobile.twitter.com", "nitter.42l.fr")                                    
autocmd DocStart (https?://)www.youtube.com js tri.excmds.urlmodify("-t", "www.youtube.com", "vid.puffyan.us")
```

I deliberately drafted it this way due to the `www` subdomain redirects for reddit and youtube. To test:

- ensure you can run `:tabopen` and `:tabopen search` from a permanent and/or temporary container and it properly redirects and only opens one tab
- ensure you can run `:tabopen` and `:tabopen search` from the new tab page and it properly redirects and only opens one tab

**Reply by bovine3dom** [reply]
Great :)

Do you still have tabopencontaineraware set to false?

**Reply by ghost** [reply]
Yes, but it was off by default anyway.

**Reply by bovine3dom** [reply]
We should probably document the settings you need somewhere. Where would you have expected to find them listed?

**Reply by ghost** [reply]
Either in the troubleshooting page or wiki, as that is linked from the tutor page. Or possibly a one liner linking to a document explaining these settings. They work now but may break in the future and I noticed some open issues here and in the temporary containers repo about interoperability. Going to do some more research first then maybe send off a PR.
